biggerquery/interactive.py

```diff
@@ -0,0 +1,785 @@
+            sql=sql,
+            table_name=table_name,
+            partitioned=partitioned,
+            custom_run_datetime=custom_run_datetime)
+
+    def write_append(self, table_name, sql, partitioned=True, custom_run_datetime=None, operation_name=None):
+        return self._run_write_operation(
+            operation_name=operation_name,
+            method=self._dataset_manager.write_append,
+            sql=sql,
+            table_name=table_name,
+            partitioned=partitioned,
+            custom_run_datetime=custom_run_datetime)
+
+    def write_tmp(self, table_name, sql, custom_run_datetime=None, operation_name=None):
+        return self._run_write_operation(
+            operation_name=operation_name,
+            method=self._dataset_manager.write_tmp,
+            sql=sql,
+            table_name=table_name,
+            custom_run_datetime=custom_run_datetime)
+
+    def collect(self, sql, custom_run_datetime=None, operation_name=None):
+        return self._run_write_operation(
+            operation_name=operation_name,
+            method=self._dataset_manager.collect,
+            sql=sql,
+            custom_run_datetime=custom_run_datetime)
+
+    def create_table(self, create_query, operation_name=None):
+        if self._operation_name == operation_name or self._operation_name is None:
+            return self._results_container, self._dataset_manager.create_table(create_query=create_query)
+
+    def load_table_from_dataframe(self, table_name, df, partitioned=True, custom_run_datetime=None, operation_name=None):
+        if self._should_peek_operation_results(operation_name):
+            return df
+        elif self._should_run_operation(operation_name):
+            return self._dataset_manager.load_table_from_dataframe(
+                table_name=table_name,
+                df=df,
+                custom_run_datetime=custom_run_datetime,
+                partitioned=partitioned)
+
+    @property
+    def dt(self):
+        return self._dataset_manager.runtime_str
+
+    @property
+    def extras(self):
+        return self._dataset_manager.extras
+
+    @property
+    def _result(self):
+        return self._results_container
+
+    @property
+    def client(self):
+        return self._dataset_manager.client
+
+    def _collect_select_result_to_pandas(self, sql):
+        sql = sql if 'limit' in sql.lower() else sql + '\nLIMIT {}'.format(str(self._peek_limit))
+        return self._dataset_manager.collect(sql)
+
+    def _run_write_operation(self, operation_name, method, sql, *args, **kwargs):
+        if self._should_peek_operation_results(operation_name):
+            result = self._collect_select_result_to_pandas(sql)
+            self._results_container.append(result)
+            return result
+        elif self._should_run_operation(operation_name):
+            return method(*args, sql=sql, **kwargs)
+        else:
+            return pd.DataFrame()
```

Comment:
This line is not doing any form of `write`

Reply:
Returning empty pandas makes a bit easier to handle peeking inside component. For example:
```python
def component(ds):
    some_data = ds.collect(...)
    for row in some_data:
        ...
    ds.write_truncate(operation_name='write')
```

When you peek 'write' operation, you don't expect to get None from collect.